stdlib/knative/src/main/ballerina/src/knative/annotation.bal

```diff
@@ -0,0 +1,232 @@
+# + dependsOn - Services this deployment depends on.
+# + imagePullSecrets - Image pull secrets.
+# + containerConcurrency - concurent request handle by one container instance
+# + timeoutSeconds - max time the instance is allowed for responding to a request
+# + port - containerPort value for Knative service
+public type ServiceConfiguration record{|
+        *Metadata;
+        string dockerHost?;
+        string dockerCertPath?;
+        string registry?;
+        string username?;
+        string password?;
+        string baseImage?;
+        string image?;
+        boolean buildImage = true;
+        boolean push = false;
+        FileConfig[] copyFiles?;
+        boolean singleYAML = true;
+        string namespace?;
+        int replicas = 1;
+        boolean|ProbeConfiguration livenessProbe = false;
+        boolean|ProbeConfiguration readinessProbe = false;
+        ImagePullPolicy imagePullPolicy = IMAGE_PULL_POLICY_IF_NOT_PRESENT;
+        map<string|FieldRef|SecretKeyRef|ResourceFieldRef|ConfigMapKeyRef> env?;
+        map<string> podAnnotations?;
+        PodTolerationConfiguration[] podTolerations?;
+        string[] dependsOn?;
+        string[] imagePullSecrets?;
+        int containerConcurrency=100;
+        int timeoutSeconds=60;
+        int port=8080;
+|};
+public const annotation ServiceConfiguration Service on source service, source function, source listener;
+# Knative Horizontal Pod Autoscaler configuration
+#
+# + minReplicas - Minimum number of replicas.
+# + maxReplicas - Maximum number of replicas.
+public type PodAutoscalerConfig record {|
+    *Metadata;
+    int minReplicas?;
+    int maxReplicas?;
+|};
+# @knative:HPA annotation to configure horizontal pod autoscaler yaml.
+public const annotation PodAutoscalerConfig HPA on source service, source function;
+# Knative secret volume mount.
+#
+# + mountPath - Mount path.
+# + readOnly - Is mount read only. Default is `true`.
+# + data - Paths to data files as an array.
+public type Secret record {|
+    *Metadata;
+    string mountPath;
+    boolean readOnly = true;
+    string[] data;
+|};
+#Secret volume mount configurations for kubernetes.
+#
+# + secrets - Array of [Secret](kubernetes.html#Secret)
+public type SecretMount record {|
+    Secret[] secrets;
+|};
+# @knative:Secret annotation to configure secrets.
+public const annotation SecretMount Secret on source service, source function;
+# Knative Config Map volume mount.
+#
+# + mountPath - Mount path.
+# + readOnly - Is mount read only. Default is `true`.
+# + data - Paths to data files.+ port - port value for the containerPort
+public type ConfigMap record {|
+    *Metadata;
+    string mountPath;
+    boolean readOnly = true;
+    string[] data;
+|};
+# Secret volume mount configurations for kubernetes.
+#
+# + conf - path to ballerina configuration file
+# + configMaps - Array of [ConfigMap](kubernetes.html#ConfigMap)
+public type ConfigMapMount record {|
+    string conf;
```

Comment:
conf does not make sense. Shall we change to "confPath" or "configPath"?

Reply:
We will keep the conf as same, we are following the Kubernetes annotations here also.